botocove/cove_host_account.py

```diff
@@ -110,11 +111,97 @@ def _resolve_target_accounts(self, target_ids: Optional[List[str]]) -> Set[str]:
             # No target_ids passed
             target_accounts = self._gather_org_assume_targets()
         else:
-            # Specific list of IDs passed
-            target_accounts = set(target_ids)
+            target_accounts, target_ous = self._validate_target_accounts(target_ids)
+            target_accounts_from_ous = self._get_all_accounts_by_organization_units(
+                target_ous
+            )
+            target_accounts.extend(target_accounts_from_ous)
+
+            target_accounts = set(target_accounts)
 
         return target_accounts - validated_ignore_ids
 
+    def _validate_target_accounts(
+        self, target_ids: Optional[List[str]]
+    ) -> Tuple[List[str], List[str]]:
+
+        target_accounts = list()
+        target_ous = []
+
+        for target_id in target_ids:
+            if not isinstance(target_id, str):
+                raise TypeError("All target_id list entries must be strings")
+            if not re.match(r"^\d{12}$", target_id):
+                if not re.match(r"^ou-[0-9a-z]{4,32}-[a-z0-9]{8,32}$", target_id):
+                    raise TypeError(
+                        f"target_ids entry is neither an aws account nor an ou: {target_id}"
+                    )
+                else:
+                    target_ous.append(target_id)
+            else:
+                target_accounts.append(target_id)
+
+        return target_accounts, target_ous
+
+    def _get_all_accounts_by_organization_units(
+        self, target_ous: Optional[List[str]]
+    ) -> List[str]:
+
+        account_list = list()
+
+        for parent_ou in target_ous:
+
+            current_ou_list = []
+            self._get_all_child_ous(parent_ou, current_ou_list)
+
+            # for complete list add parent ou as well to list of child ous
+            current_ou_list.append(parent_ou)
+
+            account_list.extend(
+                self._get_accounts_by_organization_units(current_ou_list)
+            )
+
+        return account_list
+
+    def _get_all_child_ous(
+        self, parent_ou: str, ou_list: Optional[List[str]] = None
+    ) -> List[str]:
+
+        if ou_list == None:
+            ou_list = []
+
+        extraArgs = {"ChildType": "ORGANIZATIONAL_UNIT", "ParentId": parent_ou}
+
+        paginator = self.org_client.get_paginator("list_children")
+        page_iterator = paginator.paginate(**extraArgs)
+
+        for page in page_iterator:
+            ous_batch = [child["Id"] for child in page["Children"]]
+            ou_list.extend(ous_batch)
+            for ou in ous_batch:
+                self._get_all_child_ous(ou, ou_list)
+
+        return ou_list
```

Comment:
I ended up writing a test to make sure I grokked the logic here - if I'm understanding correctly since we're mutating the list on line 154 this return statement doesn't do anything, but makes the lack of return value captured on line 182 confusing. Let's stick to one approach here for depth first recursion :)

Reply:
Thanks for the hint, I removed the return statement